src/services/resource.js

```diff
@@ -626,6 +630,119 @@ module.exports = class resourceHelper {
 		}
 	}
 
+	/**
+	 * Resource Details
+	 * @method
+	 * @name getDetails
+	 * @returns {JSON} - details of resource
+	 */
+	static async getDetails(resourceId) {
+		try {
+			let result = {
+				organization: {},
+			}
+
+			const resource = await resourceQueries.findOne({
+				id: resourceId,
+			})
+
+			if (!resource?.id) {
+				return responses.failureResponse({
+					message: 'RESOURCE_NOT_FOUND',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			if (resource.blob_path) {
+				const response = await filesService.fetchJsonFromCloud(resource.blob_path)
+				if (
+					response.statusCode === httpStatusCode.ok &&
+					response.result &&
+					Object.keys(response.result).length > 0
+				) {
+					//modify the response as label value pair
+					let resultData = response.result
+
+					//get all entity types with entities
+					let entityTypes = await entityModelMappingQuery.findEntityTypesAndEntities(
+						{
+							model: resource.type,
+							status: common.STATUS_ACTIVE,
+						},
+						resource.organization_id,
+						['id', 'value', 'label', 'has_entities']
+					)
+
+					if (entityTypes.length > 0) {
+						//create label value pair map
+						const entityTypeMap = entityTypes.reduce((map, type) => {
+							if (type.has_entities && Array.isArray(type.entities) && type.entities.length > 0) {
+								map[type.value] = type.entities
+									.filter((entity) => entity.status === common.STATUS_ACTIVE)
+									.map((entity) => ({ label: entity.label, value: entity.value.toLowerCase() }))
+							}
+							return map
+						}, {})
+
+						for (let entityType of entityTypes) {
```

Comment:
@priyanka-TL Can you help understand what this leads to? Can we do it in parallel.

Also, we need a different API to list and search with pagination to get all entities for a given entity type. Rakesh has developed the same.

Reply:
This will convert as label value pair. here we are listing entities based on model. I have updated this to parallel